dotCMS/src/test/java/com/dotmarketing/business/cache/provider/h22/H22CacheTest.java

```diff
@@ -268,4 +278,105 @@ public void run() {
 
 	}
 
+	public H22Cache newCacheInstance(){
+		File dir = Files.createTempDir();
+		dir.mkdirs();
+
+		final H22Cache cache = new H22Cache(dir.getAbsolutePath());
+		try {
+			cache.init();
+		} catch(Exception e) {
+			throw new DotRuntimeException(e);
+		}
+		return cache;
+	}
+
+	/**
+	 * Given scenario: We create a cache instance then we feed it with remove-task. The cache
+	 * it-self internally decides whether or not (according to the capacity) if some task should be
+	 * executed or not asynchronously
+	 * Expected Results:  The test must match a criteria specified
+	 * within the test-case. That helps to predict a behavior.
+	 * In any case we should never ever get a RejectedExecutionException
+	 */
+	@Test
+	@UseDataProvider("toleranceTestCases")
+	public void Test_Exhaust_Thread_Pool(final ToleranceTestCase testCase) throws Exception {
```

Comment:
Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6190817)